core/src/main/java/tc/oc/pgm/kits/KitParser.java

```diff
@@ -284,6 +271,24 @@ public ItemKit parseItemKit(Element el) throws InvalidXMLException {
     return new ItemKit(slotItems, freeItems, repairTools, deductTools, deductItems, dropOverflow);
   }
 
+  public @Nullable ItemStack parseItemStack(Element el) throws InvalidXMLException {
+    ItemStack item = null;
+    switch (el.getName()) {
+      case "item":
+        item = parseItem(el, true);
+        break;
+
+      case "book":
+        item = parseBook(el);
+        break;
+
+      case "head":
+        item = parseHead(el);
+        break;
+    }
+    return item;
```

Comment:
this could use returns on each case directly, and include a default for null:
```suggestion
    switch (el.getName()) {
      case "item":
        return parseItem(el, true);
      case "book":
        return parseBook(el);
      case "head":
        return parseHead(el);
      default:
        return null;
    }
```